distributed/utils.py

```diff
@@ -683,7 +683,6 @@ def silence_logging(level, root='distributed'):
     return old
 
 
-@toolz.memoize
```

Comment:
This slightly concerns me.  I've seen systems where DNS lookups are surprisingly expensive.  Your approach here is probably safer long though, especially for longer running processes.

Reply:
DNS caching also occurs at the OS level.

I'm not familiar with this decorator, but I do not see any mention of cache flushing to remove stale entries, as with the OS dns caches the usually have a TTL (time to live) to specify when they need to be re-queried.

With the current implementation, it appears that the @toolz.memozie will never flush the cache.